pkg/controller/storageversionmigrator/storageversionmigrator.go

```diff
@@ -271,6 +271,16 @@ func (svmc *SVMController) sync(ctx context.Context, key string) error {
 		if err != nil {
 			return err
 		}
+		objRV, err := convertResourceVersionToInt(accessor.GetResourceVersion())
+		if err != nil {
+			logger.V(4).Error(err, "Unable to migrate resource since its version is not an integer", "namespace", accessor.GetNamespace(), "name", accessor.GetName(), "gvr", gvr.String())
+			return svmc.failMigration(ctx, toBeProcessedSVM, err)
+		}
+
+		if objRV > listResourceVersion {
+			logger.V(6).Info("Resource has a newer resource version than start of migration, does not need migration", "namespace", accessor.GetNamespace(), "name", accessor.GetName(), "gvr", gvr.String())
+			continue
+		}
```

Comment:
Error handling and RV handling should be at least separate commits

Reply:
Sure done, also planning on moving to this helper once it merges https://github.com/kubernetes/kubernetes/pull/134330

Reply:
Kept in this PR but moved to using the helper in places we compare the RV since this is the official helper.